lib/gas_mrpc.c

```diff
@@ -97,14 +100,21 @@ void gas_mrpc_memcpy_from_gas(struct switchtec_dev *dev, void *dest,
 
 	while (n) {
 		cmd.len = n;
-		if (cmd.len > MRPC_MAX_DATA_LEN)
-			cmd.len = MRPC_MAX_DATA_LEN;
+		if (cmd.len > MRPC_GAS_MEMCPY_MAX_SIZE)
+			cmd.len = MRPC_GAS_MEMCPY_MAX_SIZE;
```

Comment:
This should really be a second patch... If you write "also" or "and" or whatever, it probably means it's a completely different topic.